src/pages/TaskList/ui/TaskList.test.tsx

```diff
@@ -0,0 +1,102 @@
+            </Provider>,
+        );
+
+        const taskInput = (screen.getByLabelText(/create new task/i));
+        expect(taskInput).toBeInTheDocument();
+
+        fireEvent.input(taskInput, { target: { value: 'New task' } });
+        fireEvent.click(screen.getByTestId('AddTask-button'));
+
+        expect(generateRandomId).toHaveBeenCalled();
+        expect(screen.getByText('Created:14.11.2023 / 15.53.23')).toBeInTheDocument();
+        expect(screen.getByText('New task')).toBeInTheDocument();
+
+        fireEvent.click(screen.getByText('New task'));
+
+        const subTaskInput = screen.getByLabelText(/create sub task/i);
+        expect(subTaskInput).toBeInTheDocument();
+
+        fireEvent.input(subTaskInput, { target: { value: 'New_test_sub_task' } });
+        fireEvent.click(screen.getByTestId('AddTask-button-ghj43GF6'));
+
+        expect(generateRandomId).toHaveBeenCalled();
+        expect(screen.getByText('New_test_sub_task')).toBeInTheDocument();
+    });
+
+    test('Create new sub task, when task is completed', () => {
+        localStorage.setItem('tasks', JSON.stringify([{ ...testTask, status: Status.COMPLETED, subTasks: [] }]));
+
+        render(
+            <Provider store={setupStore(initialState)}>
+                <TaskList />
+            </Provider>,
+        );
+
+        expect(screen.getAllByText(Status.COMPLETED)).toHaveLength(1);
+        expect(screen.getByText('Click for more')).toBeInTheDocument();
+
+        fireEvent.click(screen.getByText('Click for more'));
+        expect(screen.queryByText('Click for more')).not.toBeInTheDocument();
+        expect(screen.getByLabelText(/create sub task/i)).toBeInTheDocument();
+
+        fireEvent.input(screen.getByLabelText(/create sub task/i), { target: { value: 'New_test_sub_task' } });
+        fireEvent.click(screen.getByTestId(`AddTask-button-${testTask.id}`));
+
+        expect(generateRandomId).toHaveBeenCalled();
+        expect(screen.getByText('New_test_sub_task')).toBeInTheDocument();
+
+        expect(screen.queryAllByText(Status.COMPLETED)).toHaveLength(0);
+        expect(screen.getAllByText(Status.IN_PROGRESS)).toHaveLength(1);
+    });
+
+    test('Change task status', () => {
+        localStorage.setItem('tasks', JSON.stringify([testTask]));
+
+        render(
+            <Provider store={setupStore(initialState)}>
+                <TaskList />
+            </Provider>,
+        );
+
+        expect(screen.getByText(Status.TO_DO)).toBeInTheDocument();
+
+        fireEvent.click(screen.getByText(testTask.title));
+        expect(screen.getByLabelText(/create sub task/i)).toBeInTheDocument();
+        expect(screen.getByText(Status.IN_PROGRESS)).toBeInTheDocument();
+
+        act(() => {
+            userEvent.click(screen.getByText(Status.TO_DO));
+        });
+        act(() => {
+            userEvent.click(screen.getByText(Status.COMPLETED));
+        });
+
+        expect(screen.getAllByText('Done:14.11.2023 / 15.53.23')).toHaveLength(2);
```

Comment:
The test has name `Change task status` but status itself is not asserted for some reason

Reply:
Fixed